Assets/HoloToolkit/SpectatorView/Scripts/Networking/SpectatorViewNetworkDiscovery.cs

```diff
@@ -0,0 +1,275 @@
+        private NewDeviceDiscovery newDeviceDiscovery;
+
+        /// <summary>
+        /// Called when the phone finds a hololens session with its marker code.
+        /// </summary>
+        public HololensSessionFoundEvent OnHololensSessionFound;
+
+        /// <summary>
+        /// Discovery starts when the component starts
+        /// </summary>
+        public bool AutoStart
+        {
+            get { return autoStart; }
+            set { autoStart = value; }
+        }
+
+        /// <summary>
+        /// Is the discovery component stopping?
+        /// </summary>
+        public bool IsStopping
+        {
+            get { return isStopping; }
+            set { isStopping = value; }
+        }
+
+        /// <summary>
+        /// Component used to detect a AR marker from the HoloLens
+        /// </summary>
+        public MarkerDetectionHololens MarkerDetectionHololens
+        {
+            get { return markerDetectionHololens; }
+            set { markerDetectionHololens = value; }
+        }
+
+        /// <summary>
+        /// Component that generates the AR codes
+        /// </summary>
+        public MarkerGeneration3D MarkerGeneration3D
+        {
+            get { return markerGeneration3D; }
+            set { markerGeneration3D = value; }
+        }
+
+        /// <summary>
+        /// Component that manages the procedure of discovering new devices (mobile)
+        /// </summary>
+        public NewDeviceDiscovery NewDeviceDiscovery
+        {
+            get { return newDeviceDiscovery; }
+            set { newDeviceDiscovery = value; }
+        }
+
+
+        private void Awake()
+        {
+#if NETFX_CORE
+            try
+            {
+			    OpenCVUtils.CheckOpenCVWrapperHasLoaded();
+            }
+            catch (Exception e)
+            {
+                Debug.LogError(e);
+                gameObject.SetActive(false);
+                return;
+            }
+#endif
+            isHost = FindObjectOfType<PlatformSwitcher>().TargetPlatform == PlatformSwitcher.Platform.Hololens;
+
+            // The client doesn't have to wait for the server to be started, but this works best if the component
+            // waits for the remaining networking bits to have warmed up,
+            // just give it a couple of seconds and then start it
+            if (!isHost)
+            {
+                Invoke("ManualStart", 3f);
+            }
+        }
+
+        /// <summary>
+        /// Starts the system in server or client mode depending on isHost
+        /// </summary>
+        public void ManualStart()
+        {
+            // Auto find components if necessary
+            if (MarkerGeneration3D == null)
+            {
+                MarkerGeneration3D = FindObjectOfType<MarkerGeneration3D>();
+            }
+
+            if (MarkerDetectionHololens == null)
+            {
+                MarkerDetectionHololens = FindObjectOfType<MarkerDetectionHololens>();
```

Comment:
Yeah, these should normally get set in the Inspector (and they do here).

![image](https://user-images.githubusercontent.com/3580640/39555504-9f6360ea-4e2e-11e8-9151-93e0ef203338.png)

Looks like this call is more of a "worst case, somebody forgot to set this in the inspector. Instead of failing, let's go find it."

Reply:
this only occurs if the user did not populate the field in the inspector